src/mimefactory.rs

```diff
@@ -458,20 +458,45 @@ impl<'a> MimeFactory<'a> {
             self.from_addr.clone(),
         );
 
+        let undisclosed_recipients = match &self.loaded {
+            Loaded::Message { chat } => chat.typ == Chattype::Broadcast,
+            Loaded::Mdn { .. } => false,
+        };
+
         let mut to = Vec::new();
-        for (name, addr) in self.recipients.iter() {
-            if name.is_empty() {
-                to.push(Address::new_mailbox(addr.clone()));
-            } else {
-                to.push(Address::new_mailbox_with_name(
-                    name.to_string(),
-                    addr.clone(),
-                ));
+        if undisclosed_recipients {
+            to.push(Address::new_group(
+                "hidden-recipients".to_string(),
```

Comment:
In case someone else reviews: I was a bit confused on first sight because link2xt changed `hidden-recipients: ;` to `hidden-recipients` here, but this works because he wraps it in `Address::new_group(..., Vec::new())`, which means exactly the same as `hidden-receipients: ;`: A group of email addresses called `hidden-recipients` that contains 0 addresses.